bridge_adaptivity/module/forms.py

```diff
@@ -130,42 +125,100 @@ def clean(self):
 
-    groups = forms.ModelMultipleChoiceField(
-        label="Choose groups to add into this course:",
-        queryset=CollectionGroup.objects.filter(course__isnull=True),
-        widget=forms.CheckboxSelectMultiple()
-    )
-
     def save(self, **kwargs):
+        """
+        Save change that related to add groups to course.
+        """
         group_ids = [group.id for group in self.cleaned_data['groups']]
-        CollectionGroup.objects.filter(id__in=group_ids).update(course=self.course)
+        ModuleGroup.objects.filter(id__in=group_ids).update(course=self.course)
+
+
+class CollectionOrderForm(ModelForm):
+    """
+    Add collection in group form.
+    """
+
+    grading_policy_name = forms.ChoiceField(
+        choices=((k, v) for k, v in GRADING_POLICY_NAME_TO_CLS.items()),
+        required=True,
+        widget=PolicyChoiceWidget
+    )
 
+    class Meta:
+        """
+        Metaclass for CollectionOrderForm.
+        """
 
-class AddCollectionGroupForm(forms.Form):
-    """Add collection in group form."""
+        model = CollectionOrder
+        fields = (
+            'slug',
+            'collection',
+            'engine',
+            'grading_policy_name',
+            'strict_forward',
+            'ui_option',
+            'ui_next'
+        )
+        labels = {'ui_option': _('UI Option'), 'ui_next': _('Additional NEXT Button')}
 
     def __init__(self, *args, **kwargs):
         self.user = kwargs.pop('user')
         self.group = kwargs.pop('group')
+        read_only = kwargs.pop('read_only') if 'read_only' in kwargs else False
         super().__init__(*args, **kwargs)
-        self.fields['collections'].queryset = self.fields['collections'].queryset.filter(
+        self.fields['collection'].queryset = self.fields['collection'].queryset.filter(
             owner_id=self.user.id
-        ).exclude(collection_groups=self.group)
+        )
+        if read_only:
+            self.fields['collection'].widget = HiddenInput()
+            self.fields['collection'].widget.attrs['readonly'] = read_only
 
-    collections = forms.ModelMultipleChoiceField(
-        label="Choose colections to add into this group:",
-        queryset=Collection.objects.filter(),
-        widget=forms.CheckboxSelectMultiple()
-    )
+    def clean(self):
+        """
+        Make cleaned_data from Query.
+        """
+        super().clean()
+        engine, policy = self.cleaned_data.get('engine'), self.cleaned_data.get('grading_policy_name')
+
+        engine_cls = engine.engine_driver
+        policy_cls = GRADING_POLICY_NAME_TO_CLS.get(policy)
+
+        if policy_cls is None:
+            raise forms.ValidationError({'grading_policy_name': ['Not correct policy']})
+
+        required_engine = policy_cls.require.get('engine')
+
+        if required_engine and not isinstance(engine_cls, required_engine):
+            required_engine_names = ", ".join([e.__name__.strip('Engine') for e in required_engine])
```

Comment:
Maybe we need to change reqired_engine to required_engines. And write the variable 'e' more clearly ('engine').

Reply:
Now policy requires only one engine `required_engine_names` looks like an odd parameter.